File: eulers/1/hs/Main.hs
main :: IO ()
main = putStrLn . show . sumIntsDivisBy3or5To $ 999

isDivisBy :: Int -> Int -> Bool
isDivisBy x y = mod x y == 0

isDivisBy3or5 :: Int -> Bool
isDivisBy3or5 x = (isDivisBy x 3) || (isDivisBy x 5)

sumIntsDivisBy3or5To :: Int -> Int
sumIntsDivisBy3or5To x = sum $ filter isDivisBy3or5 [0..x]

File: eulers/1/jl/main.jl
function sum_ints_divis_by_3_or_5(n::Int)
	is_divis_by_3_or_5 = x -> x % 3 == 0 || x % 5 == 0
	return sum(filter(is_divis_by_3_or_5, 1:n))
end

println(sum_ints_divis_by_3_or_5(999))

